front/components/agent_builder/AgentBuilderFormContext.tsx

```diff
@@ -71,12 +67,14 @@ const dataSourceViewSelectionConfigurationSchema = z.object({
   tagsFilter: tagsFilterSchema,
 });
 
+export const dataSourceConfigurationSchema = z.record(
+  z.string(),
+  dataSourceViewSelectionConfigurationSchema
+);
+
 const searchActionConfigurationSchema = z.object({
   type: z.literal("SEARCH"),
-  dataSourceConfigurations: z.record(
-    z.string(),
-    dataSourceViewSelectionConfigurationSchema
-  ),
+  dataSourceConfigurations: dataSourceConfigurationSchema,
```

Comment:
`nit`: we could have a base schema with:
```
dataSourceConfigurations: dataSourceConfigurationSchema,
```
since all configs actually have that field?

Reply:
Am I correct that this will be solved when we only have `mcpServerConfigurationSchema`? 🤔

Reply:
I'm actually thinking, would it be easier if we exclude action from agent builder form, and have `actions` as state instead, and we use RHF for action configuration form but not dataSourceConfigurations ? 🤔